### PR TITLE
fix(rds): userpool & iam e2e tests

### PR DESCRIPTION
**Pull request by AaronZyLee**
#### Description of changes
Fixed failing e2e tests

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by phani-srikar — APPROVED** [review]
Let's merge this after the merge from main PR, since it avoids refreshing that one and running full E2E again.

**Review by phani-srikar — APPROVED** [review]
Good to merge after these E2Es pass.

**Review by palpatim — APPROVED** [review]

**Pull request merged**
